### PR TITLE
[hotfix] Fixes the TimestampedInputSplit.EOS comparison.

### PR DESCRIPTION
**Pull request by kl0u**

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/TimestampedFileInputSplitTest.java [hunk 1]
```diff
@@ -90,4 +112,60 @@ public void testIllegalArgument() {
...
+			new TimestampedFileInputSplit(11, 1, new Path("test/test3"), 0, 100, null);
+
+		TimestampedFileInputSplit eos = TimestampedFileInputSplit.EOS;
+
+		Queue<TimestampedFileInputSplit> pendingSplits = new PriorityQueue<>(10, new Comparator<TimestampedFileInputSplit>() {
+			@Override
+			public int compare(TimestampedFileInputSplit o1, TimestampedFileInputSplit o2) {
+				return o1.compareTo(o2);
+			}
+		});
```
Comment (mxm) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/TimestampedFileInputSplitTest.java]: `TimestampedFileInputSplit` is already `Comparable`. So you can skip the Comparator here.

**Comment by mxm** [comment]
I think we should get rid of the magic `EOF` split. We should simply set a flag in the reader to stop reading. Until then, this looks like a good fix.

**Comment by kl0u** [comment]
@mxm I agree and I am planning to do it as soon as we fix some other design issues with the ContinuousFileReadOperator that concern rescaling. Thanks for the review.

**Comment by kl0u** [comment]
Thanks @mxm !

**Pull request closed without merging**